code/modules/mob/mob_grab.dm

```diff
@@ -332,7 +334,84 @@
+								G.state = GRAB_PASSIVE
+							if(prob(15))
+								visible_message("<span class='danger'>[assailant] has [H] grasped by their [hairstyle], tugging on it roughly!</span>")
+								H.apply_damage(9, HALLOSS)
+								G.state = GRAB_AGGRESSIVE
+							if(prob(3))
+								visible_message("<span class='danger'>[assailant] has [H] grasped by their [hairstyle], tugging their hair out violently</span>")
+								H.apply_damage(15, HALLOSS)
+								H.h_style = H.species.default_h_style
+								H.update_hair()
+								G.state = GRAB_AGGRESSIVE
+
+						if(H.h_style in hairsemisafe)
+							if(prob(55))
+								visible_message("<span class='danger'>[assailant] has [H] grasped by their [hairstyle], tugging on it tightly!</span>")
+								H.apply_damage(5, HALLOSS)
+								G.state = GRAB_PASSIVE
+							if(prob(10))
+								visible_message("<span class='danger'>[assailant] has [H] grasped by their [hairstyle], tugging on it roughly!</span>")
+								H.apply_damage(7, HALLOSS)
+								G.state = GRAB_PASSIVE
+							if(prob(5))
+								visible_message("<span class='danger'>[assailant] has [H] grasped by their [hairstyle], tugging a clump of hair out violently</span>")
+								H.apply_damage(15, HALLOSS)
+								G.state = GRAB_AGGRESSIVE
+
+						if(H.h_style in hairsortasafe || cathair)
+							if(prob(40))
+								visible_message("<span class='danger'>[assailant] has [H] grasped by their [hairstyle], tugging on it tightly!</span>")
+								H.apply_damage(5, HALLOSS)
+								G.state = GRAB_PASSIVE
+							if(prob(15))
+								visible_message("<span class='danger'>[assailant] has [H] grasped by their [hairstyle], tugging on it roughly!</span>")
+								H.apply_damage(3, HALLOSS)
+								G.state = GRAB_AGGRESSIVE
+							if(prob(15))
+								visible_message("<span class='danger'>[assailant] has [H] grasped by their [hairstyle], tugging a clump of hair out violently</span>")
+								H.apply_damage(15, HALLOSS)
+								G.state = GRAB_AGGRESSIVE
+
+						if(H.h_style in hairunsafe)
+							if(prob(80))
+								visible_message("<span class='danger'>[assailant] has [H] grasped by their [hairstyle], tugging on it tightly!</span>")
+								H.apply_damage(10, HALLOSS)
+								G.state = GRAB_PASSIVE
+							if(prob(30))
+								visible_message("<span class='danger'>[assailant] attempted to grasp [H] by their [hairstyle], however they lose there grip!</span>")
+								G.state = GRAB_PASSIVE
+							if(prob(5))
+								visible_message("<span class='danger'>[assailant] has [H] grasped by their [hairstyle], tugging their hair out violently</span>")
+								H.apply_damage(15, HALLOSS)
+								H.h_style = H.species.default_h_style
+								H.update_hair()
+								G.state = GRAB_AGGRESSIVE
+
+						if(H.h_style in skrelltentacles)
+							if(prob(95))
+								visible_message("<span class='danger'>[assailant] has [H] grasped by their [hairstyle], however suddenly it slips from  [assailant]'s hand!</span>")

```

Comment:
as I've recently found out, using the hair style name is a horrible idea in any IC context.

"Urist McCqc grabs Mary Sueflake by the Ponytail 2!" Doesn't have a good ring to it. I got around this by renaming literally all the hairs in my codebase, which I could only do because I got rid of most of them. Perhaps a hair_desc var on the hair datum would work.